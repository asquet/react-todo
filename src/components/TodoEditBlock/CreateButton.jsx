import React from 'react';

function CreateButton({ onClick }) {
  return (
    <button
      className="btn btn-primary"
      onClick={(ev) => {
        ev.preventDefault();
        onClick();
      }}
    >Add todo </button>
  );
}

CreateButton.propTypes = {
  onClick: React.PropTypes.func.isRequired
};

export default CreateButton;
