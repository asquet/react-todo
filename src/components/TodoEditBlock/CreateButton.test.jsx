import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import CreateButton from './CreateButton';


it(
  'renders without crashing', () => {
    const component = renderer.create(
      <CreateButton
        onClick={() => {
        }}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  }
);

it(
  'responds to click', () => {
    const onClick = jest.fn();
    const rendered = shallow(<CreateButton onClick={onClick} />);
    rendered.simulate('click', { preventDefault: () => {} });

    expect(onClick.mock.calls.length).toBe(1);
  }
);
