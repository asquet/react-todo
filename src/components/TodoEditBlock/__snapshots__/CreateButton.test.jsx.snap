exports[`test renders without crashing 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}>
  Add todo 
</button>
`;
